### ISSUE TITLE
MenuBar: Allow non menu items inside the menu

### ISSUE THREAD

**Issue by omzy** [primefaces/primevue#4480]
### Describe the feature you would like to see added

Currently, the `model` prop takes an array of menu items. These items will render the menu as a `ul > li` list.

I would like to be able to include other, non menu items inside this menu. See demo here: https://stackblitz.com/edit/prncz5-kcxnkk

In my demo, I have used the `end` slot, which is fine on desktop, but on mobile devices these items appear at the end of the menu bar. Ideally, I would like these items to appear inside the main (hidden) menu.

The ideal markup I would like to have is something like this:

```
<div class="p-menubar p-component">
    <a class="p-menubar-button"><svg>(menu button)</svg></a>

    <div class="main">
        <ul class="p-menubar-root-list>
            <li class="p-menuitem">Menu item</li>
            <li class="p-menuitem">...</li>
        </ul>

        <div class="custom-content">
            <button>1</button>
            <some other content>
        </div>
    </div>
</div>
```

So basically, clicking `p-menubar-button` will toggle `div.main`.

**Comment by tugcekucukoglu** [comment]
Issue tracker is used for defects only as part of our commitment to quality and continuous improvement in all areas. Enhancements are collected as valuable community feedback and managed internally so moving this enhancement ticket to our internal project management backlog. 

For enhancement requests and other topics, you can open a new topic in [PrimeVue discussions](https://github.com/orgs/primefaces/discussions/categories/primevue).
